### PR TITLE
add read only call sc endpoint

### PR DESCRIPTION
**Pull request by Thykof**
## TODO
- [x] add endpoint
- [x] add robot framework test

## Delivery Quality Checklist

- [ ] **Breaking Changes in API:**
  Does this PR introduce breaking changes in the API?
    - If yes, have you considered making it backward compatible?
    - If backward compatibility is not considered, set the "breaking-change" label.

- [ ] **Changelog:**
  - [ ] For bugfix PR, set the "bugfix" label
  - [ ] If this change should not appear in changelog, use "ignore-for-changelog" label

- [ ] **Version Update Handling:**
  Have you ensured that the version update by user is handled correctly?

- [ ] **PR Dependency:**
  Does this PR depend on another PR?
    - If yes, is it necessary for the dependency to be released prior to merging this one?

- [ ] **Documentation:**
  - [ ] Are any necessary changes made to user-facing documentation?
  - [ ] Confirm that API documentation is updated with any relevant changes.
  - [ ] Check that README and other documentation files are accurate and current.

## QA

```curl
curl --location 'http://station.massa/cmd/read-only/callsc' \
--header 'Content-Type: application/json' \
--data '{
  "nickname": "fullpower",
  "name": "rien",
  "at": "AU1MPDRXuR22mwYDFCeZUDgYjcTAF1co6xujx2X6ugoHeYeGY3B5",
  "args": "",
  "coins": "1000446777777"
}'
```

### CODE REVIEW

File: int/api/cmd/utils.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+import (
+	"strconv"
+
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/massalabs/station/api/swagger/server/models"
+	"github.com/massalabs/station/api/swagger/server/restapi/operations"
+	sendOperation "github.com/massalabs/station/pkg/node/sendoperation"
+)
+
+// amountToUint64 converts fee to uint64.
```
Comment (Thykof) [inline int/api/cmd/utils.go:12]:
```suggestion
// amountToUint64 converts the given amount to uint64.
```

File: int/api/cmd/utils.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+			&models.Error{
+				Code:    errorInvalidArgs,
+				Message: "Error during amount conversion: " + err.Error(),
+			})
+	}
+
+	return amountString, nil
+}
+
+// CreateReadOnlyResult Converts an instance of sendOperation.ReadOnlyResult struct to  models.ReadOnlyResult struct
```
Comment (Thykof) [inline int/api/cmd/utils.go:50]:
```suggestion
// CreateReadOnlyResult Converts an instance of sendOperation.ReadOnlyResult struct to models.ReadOnlyResult struct
```

File: int/api/cmd/read_only_call_sc.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+		return operations.NewCmdReadOnlyCallSCBadRequest().WithPayload(
+			&models.Error{
+				Code:    errorInvalidNickname,
+				Message: "Error during wallet fetch: " + err.Error(),
+			})
+	}
+
+	if params.Body.Fee == "" {
+		params.Body.Fee = "0"
+	}
```
Comment (thomas-senechal) [inline int/api/cmd/read_only_call_sc.go:45]: What about setting the default value directly in the swagger file ? 
That would be simpler than setting default values in the code like that 🤔

File: int/api/cmd/read_only_call_sc.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	if errResponse != nil {
+		return errResponse
+	}
+
+	result, err := sendOperation.ReadOnlyCallSC(
+		params.Body.At,
+		params.Body.Name,
+		args,
+		coins,
+		string(params.Body.Fee),
```
Comment (thomas-senechal) [inline int/api/cmd/read_only_call_sc.go:57]: If `Fee` is already of type `string`, why make it as a string again ? 🤔

Comment (Thykof) [reply int/api/cmd/read_only_call_sc.go:57]: Fee is of type Amount which is type string but golang don't want implicit cast

File: int/api/cmd/read_only_call_sc.go (hunk 3 above)
Comment (Thykof) [reply int/api/cmd/read_only_call_sc.go:45]: you're right

Comment (Thykof) [reply int/api/cmd/read_only_call_sc.go:45]: but it doesn't work

**Pull request merged**